examples/addr2line.rs

```diff
@@ -0,0 +1,198 @@
+extern crate gimli;
+extern crate getopts;
+
+use std::env;
+use std::str::FromStr;
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+    let mut opts = getopts::Options::new();
+    opts.optopt("e", "exe", "Set the input file name (default is a.out)", "<executable>");
+
+    let matches = match opts.parse(&args[1..]) {
+        Ok(m) => m,
+        Err(f) => { panic!(f.to_string()); }
+    };
+
+    let file_path = match matches.opt_str("e") {
+        Some(f) => f,
+        None => "a.out".to_string()
+    };
+
+    let file = obj::open(&file_path);
+    if obj::is_little_endian(&file) {
+        do_work::<gimli::LittleEndian>(&file, &matches);
+    } else {
+        do_work::<gimli::BigEndian>(&file, &matches);
+    }
+}
+
+fn do_work<Endian>(file: &obj::File, matches: &getopts::Matches)
+    where Endian: gimli::Endianity
+{
+    let addrs: Vec<u64> = matches.free.iter().map(|x| u64::from_str_radix(x,16).expect("Failed to parse")).collect();
```

Comment:
Yes.

We need some sort of parsing function.  Ideally one that could handle at a minimum decimal numbers and hex with a 0x prefix.  The built in from_str methods don't :/